src/eve/visitors.py

```diff
@@ -102,6 +107,11 @@ def apply(cls, tree, init_var, foo, bar=5, **kwargs):
 
     """
 
+    previsitors: Optional[Tuple[PREVISITOR]] = None
+
+    def __init__(self, *args: PREVISITOR):
+        self.previsitors = cast(Optional[Tuple[PREVISITOR]], args)
+
```

Comment:
Why it is needed to have a previsitors attribute in the instances? Wouldn't it be enough to have a class variable with the previsitors used in the specific class? Something like:

```suggestion
    previsitors: ClassVar[Optional[Tuple[PREVISITOR]]] = None

```